### PR TITLE
Might fix issue 58

### PR DESCRIPTION
**Pull request by lemire**
Fixes https://github.com/CarlVerret/csFastFloat/issues/58

### CODE REVIEW

File: csFastFloat/FastFloatParser.cs [hunk 1]
```diff
@@ -24,16 +24,13 @@ public static float exact_power_of_ten(long power)
...
       ulong word = am.mantissa;
       word |= (ulong)(uint)(am.power2) << FloatBinaryConstants.mantissa_explicit_bits;
       word = negative ? word | ((ulong)(1) << FloatBinaryConstants.sign_index) : word;
-
-      float d = 0;
-      Unsafe.Copy(ref d, &word);
-
-      return d;
+      uint truncated_word = (uint) word;
+      return BitConverter.Int32BitsToSingle(truncated_word);
```
Comment (gfoidl) [inline csFastFloat/FastFloatParser.cs:33]:
```suggestion
      return BitConverter.Int32BitsToSingle((int)truncated_word);
```
To fix the build-error.

Or 
```c#
int truncated_word = (int)word;
return BitConverter.Int32BitsToSingle(truncated_word);
```

**Comment by lemire** [comment]
@gfoidl So hopefully you see what I had in mind. It still leaves open bug possibilities with big endian systems *because* we do not test. We are smart people but even smart people can't predict what could go wrong on systems they do not have access to.

**Comment by gfoidl** [comment]
Yep, that's fine. My mind was too knotty here...

In #49 care needs to be taken at

https://github.com/CarlVerret/csFastFloat/blob/c1084002607017e8caff3b7a61610b6b4655317e/csFastFloat/Utils/Utils.cs#L38-L42
https://github.com/CarlVerret/csFastFloat/blob/c1084002607017e8caff3b7a61610b6b4655317e/csFastFloat/Utils/Utils.cs#L55-L59

The `Unsafe.ReadUnaligned` is like a `reinterpret_cast<ulong*>` from an `unsigned char` source in C++ terms.

**Comment by lemire** [comment]
@gfoidl  I am aware, but this is guarded... See...

```C#
    unsafe internal static bool is_made_of_eight_digits_fast(ulong val)
    {
      // We only enable paths depending on this function on little endian
      // platforms (it happens to be effectively nearly everywhere).
      return BitConverter.IsLittleEndian && (((val & 0xF0F0F0F0F0F0F0F0) |
               (((val + 0x0606060606060606) & 0xF0F0F0F0F0F0F0F0) >> 4)) ==
              0x3333333333333333);
    }
```

So `is_made_of_eight_digits_fast` will always return false on big endian systems.

**Review by gfoidl — APPROVED** [review]
Then it's perfect, simple and clean 👍

**Comment by lemire** [comment]
Merging with approval from @gfoidl

**Pull request merged**